[4.0] Deletes some specific cpanel strings for help and system presets and creates mod_menu strings with same value

infograf768 opened the pull request:
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/27072

### Summary of Changes
As title says. 
When a system or help preset is placed in the sidebar menu position, whether in an Administrator Menu Module (or an Administrator Dashboard Module which is not advised as we would have a bad display) , and the page loaded is Cpanel, no problem.
But when we switch page for example to Articles Manager, the titles of these presets are not loaded and therefore untranslated.

To solve the issue this PR changes the strings to mod_menu strings and corrects presets 


### Testing Instructions
Create system and help presets Administrator Menu modules and place them in menu position. 


### Before patch
COM_CPANEL_DASHBOARD_HELP_FIND_HELP="Find Help"
COM_CPANEL_DASHBOARD_HELP_GET_INVOLVED="Get Involved"
COM_CPANEL_DASHBOARD_HELP_MEET_JOOMLERS="Meet Joomlers"
COM_CPANEL_DASHBOARD_HELP_NEWS="News"
COM_CPANEL_DASHBOARD_HELP_RESOURCES="Resources"
COM_CPANEL_SYSTEM_SETUP="Setup"

are untranslated See https://github.com/joomla/joomla-cms/issues/27072

### After patch
All is fine
<img width="338" alt="Screen Shot 2019-11-17 at 09 19 02" src="https://user-images.githubusercontent.com/869724/69004963-5733c900-091b-11ea-8dcb-bb6d186268b2.png">

@Quy @brianteeman

infograf768 committed:
[4.0] Always load cpanel strings when system or help presets are in menu
position

Quy commented:
Still untranslated with Help.

![27092](https://user-images.githubusercontent.com/368084/69016564-3d62a800-0954-11ea-902c-387952abb932.jpg)

infograf768 commented:
You are correct. This happens when no default or alternate menu is also published in menu position.
Would be totally idiotic to do that but, nevertheless, looking into it

infograf768 committed:
Move loading cpanel strings out of conditional

infograf768 commented:
@Quy Found the reason. Please test again.

Quy commented:
I have tested this item :white_check_mark: successfully on ac8d6b839941dc07ab28248d30f71c587cd9e387<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27092">issues.joomla.org/tracker/joomla-cms/27092</a>.</sub>

mbabker commented on administrator/modules/mod_menu/Menu/CssMenu.php:
```diff
@@ -477,6 +477,10 @@ protected function preprocess($parent)
 			}
 
+			// Always load cpanel strings when system or help presets are in menu position
+			$language->load('com_cpanel', JPATH_ADMINISTRATOR, null, false, true) ||
```
This looks like it is loading the com_cpanel language strings on every iteration of this foreach loop if the code reaches this point.  Calls to `Language::load()` are not the most efficient thing.  Can this be moved outside the loop?

infograf768 replied:
> Can this be moved outside the loop?

@mbabker Yes indeed, it should.
But before I do, what do you mean by ` Calls to Language::load() are not the most efficient thing.`?

mbabker replied:
Last time I benchmarked it the `Language::load()` method didn’t have the greatest performance, especially when dealing with already loaded resources.  So when possible it’s a good idea to avoid repeated calls to that method with the same parameters.

SharkyKZ commented:
Or maybe the preset should not be using `com_cpanel` strings?

infograf768 commented:
> Or maybe the preset should not be using com_cpanel strings?

Only other way would be to add them to en-GB.ini
More confusing imho. Let's keep the load, but out of the loop. Just waiting for Michael response.

Note: I also found some issue with lang overrides for dashboard.

infograf768 committed:
out of loop

SharkyKZ commented:
Or add them to `mod_menu` language file so it uses its own strings like it should.

SharkyKZ commented on administrator/modules/mod_menu/Menu/CssMenu.php:
```diff
@@ -485,6 +485,10 @@ protected function preprocess($parent)
 
+		// Always load cpanel strings when system or help presets are in menu position
+		$language->load('com_cpanel', JPATH_ADMINISTRATOR, null, false, true) ||
+		$language->load('com_cpanel', JPATH_ADMINISTRATOR . '/components/com_cpanel', null, false, true);
```
Last 3 arguments can be omitted as they're the same as function defaults.

infograf768 commented:
> Or add them to mod_menu language file so it uses its own strings like it should.

I guess this is the best solution. On it.

infograf768 committed:
Merge remote-tracking branch 'upstream/4.0-dev' into 4.0_untranslated_cpanel_strings

infograf768 committed:
Moving and renaming system and help presets parent cpanel strings to
mod_menu

infograf768 commented:
@Quy @SharkyKZ @mbabker 
Please test again. Deleted the cpanel strings concerned and created mod_menu strings with same value but with mod_menu_ constants. Modified both presets to fit.

SharkyKZ commented:
I have tested this item :white_check_mark: successfully on 438c4a76a19b5b7dcd4b45ef72e5f8d9e3e83c8f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27092">issues.joomla.org/tracker/joomla-cms/27092</a>.</sub>

Quy commented:
I have tested this item :white_check_mark: successfully on 0be34ed47c81f3c17ac273b2159e3fac2729dc1d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27092">issues.joomla.org/tracker/joomla-cms/27092</a>.</sub>

Quy commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27092">issues.joomla.org/tracker/joomla-cms/27092</a>.</sub>

brianteeman commented:
As the PR has changed since the first post could you please update the title and description

brianteeman commented:
Thank

wilsonge commented:
@infograf768 can you fix conflicts here please

infograf768 commented:
Sorry, can't fix conflicts on this branch.
I have to make a new PR.

infograf768 committed:
Merge remote-tracking branch 'upstream/4.0-dev' into
4.0_untranslated_cpanel_strings

Conflicts:
	administrator/language/en-GB/en-GB.com_cpanel.ini
	administrator/language/en-GB/en-GB.mod_menu.ini

infograf768 commented:
Closing

infograf768 commented:
Please see https://github.com/joomla/joomla-cms/pull/27146
